src/coreclr/src/debug/createdump/crashinfomac.cpp

```diff
@@ -78,7 +36,14 @@ CrashInfo::EnumerateAndSuspendThreads()
     thread_act_port_array_t threadList;
     mach_msg_type_number_t threadCount;
 
-    kern_return_t result = ::task_threads(Task(), &threadList, &threadCount);
+    kern_return_t result = ::task_suspend(Task());
```

Comment:
My only concern with this one is:

```
 The function does not return until all of the task's threads have been suspended.
```

which can happen if there's non-atomic kernel operations on the thread (the failure case for thread_abort_safely). This won't change anything of how it works now, although I thought we'd talked about a retry cap before.

Reply:
Part of this change did remove the thread_abort_safely check/loop for individual thread suspend (in SuspendMachThread). That function doesn't actually wait until the thread is suspended so after we went through all the threads in the process, they could still be running where task_suspend blocks until they are so there isn't any changes in the process will gathering info. 

There still could be the case which I don't think is that common that task_suspend blocks forever because of some thread stuck in a kernel call.